operator/src/test/java/org/keycloak/operator/CRSerializationTest.java

```diff
@@ -0,0 +1,22 @@
+package org.keycloak.operator;
+
+import io.fabric8.kubernetes.client.utils.Serialization;
+import org.junit.jupiter.api.Test;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CRSerializationTest {
+
+    @Test
+    public void testDeserialization() {
```

Comment:
left this unit tests as it actually spots the underlying issue:
  - use a yaml source file
  - deserialize it
this is not explicitly tested by the current E2E